tests/test_consumer.py

```diff
@@ -204,6 +212,84 @@ def test_multistorage_strategy(
             strategy.join()
 
 
+@patch("snuba.consumers.consumer.InsertBatchWriter.close")
+@patch("snuba.consumers.consumer.DeadLetterStep.submit")
+@pytest.mark.parametrize(
+    "processes, input_block_size, output_block_size",
+    [
+        pytest.param(1, int(32 * 1e6), int(64 * 1e6), id="multiprocessing"),
+        pytest.param(None, None, None, id="no multiprocessing"),
+    ],
+)
+def test_multistorage_strategy_dead_letter_step(
+    mock_submit: MagicMock,
+    mock_close: MagicMock,
+    processes: Optional[int],
+    input_block_size: Optional[int],
+    output_block_size: Optional[int],
+) -> None:
+    from snuba.datasets.storages import StorageKey
+
+    clock = TestingClock()
+    broker: Broker[KafkaPayload] = Broker(MemoryMessageStorage(), clock)
+
+    topic = Topic("test-dlq")
+
+    broker.create_topic(topic, partitions=1)
+
+    producer = broker.get_producer()
+
+    # we want to raise a dummy clickhousewritererror so that we cause
+    # the multistorage collector to use the dead letter step
+    mock_close.side_effect = ClickhouseWriterError("oops", code=500, row=None)
+
+    commit = Mock()
+    storage_keys = [StorageKey.ERRORS_V2, StorageKey.TRANSACTIONS_V2]
+
+    storages = [get_writable_storage(key) for key in storage_keys]
+
+    strategy = MultistorageConsumerProcessingStrategyFactory(
+        storages,
+        10,
+        10,
+        processes,
+        input_block_size,
+        output_block_size,
+        TestingMetricsBackend(),
+        producer,
+        topic,
+    ).create(commit)
+
+    payloads = [
+        KafkaPayload(
+            None,
+            json.dumps((2, "insert", get_raw_event(),)).encode("utf8"),
+            [("transaction_forwarder", "1".encode("utf8"))],
+        ),
+        KafkaPayload(
+            None,
+            json.dumps((2, "insert", get_raw_transaction(),)).encode("utf8"),
+            [("transaction_forwarder", "0".encode("utf8"))],
+        ),
+    ]
+
+    now = datetime.now()
+
+    messages = [
+        Message(Partition(Topic("topic"), 0), offset, payload, now, offset + 1)
+        for offset, payload in enumerate(payloads)
+    ]
+
+    for message in messages:
+        strategy.submit(message)
+
+    strategy.close()
+    strategy.join()
+
+    # assert that we submit to the dead letter step for both messages
+    assert len(mock_submit.call_args_list) == 2
```

Comment:
There is a better way to test this. 
Pass a mock producer to the dead letter step and check what is produced there.